pkgs/native_assets_builder/doc/protocol/v1.md

```diff
@@ -0,0 +1,216 @@
+Build Hook Specification
+========================
+
+Version 1: As of 2024-01-24 implemented behind `--enable-experiment=native-assets`
+
+### Concepts
+
+#### Asset
+A file that is identified by an `assetId`. There may be multiple files with the
+same `assetId` with different characteristics such as `target` or `link_mode`.
+
+Note that in v1 there are _only_ native code assets, no data assets.
+
+#### AssetId
+An asset must have an `assetId`. Dart code that uses an asset, references the
+asset using the `assetId`.
+
+A package MUST prefix all assetIds it defines with: `package:<package>/`, this
+ensures assets don't conflict between packages.
+
+Conventionally, an asset referenced from `lib/src/foo.dart` in `package:foo`
+has `assetId = 'package:foo/src/foo.dart'`.
+
+#### Target
+
+A target specifies operating system and architecture.
+
+List of `platform` and `architecture` tuples supported by the
+Dart and Flutter SDKs to be specified below:
+ * `linux_x64`
+ * TODO: enumerate full list of supported platform/architecture tuples.
+
+
+
+### `build.dart`
+A package with native assets MUST define a `build.dart` file which accepts
+a `--config` option as follows:
+
+```console
+$ dart build.dart --config <build_config.yaml>
+```
+
+The `--config` option specifies location of the `build_config.yaml` file.
+
+The `build.dart` file MUST:
+ * Read the `build_config.yaml` file,
+ * Build assets using configuration from `build_config.yaml`.
+   * If `dry_run: true` in `build_config.yaml`, then this may be skipped.
+ * Write assets into `out_dir` from `build_config.yaml`.
+   * If `dry_run: true` in `build_config.yaml`, then this may be skipped.
+   * Filename are unrelated to `assetId`.
```

Comment:
What does this mean?

Reply:
The file name inside out_dir/ doesn't really matter, because in Dart code you'll always access things by assetId. So in some sence the file name is irrelevant.